app/src/test/java/com/example/tanya/godeltechchallengeandroid/ui/main/MainPresenterTest.kt

```diff
@@ -0,0 +1,105 @@
+package com.example.tanya.godeltechchallengeandroid.ui.main
+
+import com.example.tanya.godeltechchallengeandroid.RxSchedulerRule
+import com.example.tanya.godeltechchallengeandroid.domain.DomainContract
+import com.example.tanya.godeltechchallengeandroid.domain.entity.Word
+import com.nhaarman.mockito_kotlin.eq
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.verifyNoMoreInteractions
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Matchers.anyString
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
+
+class MainPresenterTest {
+
+    @Rule
+    @JvmField
+    var testSchedulerRule = RxSchedulerRule()
+
+    @Rule
+    @JvmField
+    var mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+    @Mock
+    private lateinit var countWordsUseCase: DomainContract.CountWordsUseCase
+
+    @Mock
+    private lateinit var view: MainContract.View
+
+    @Mock
+    private lateinit var testWords: List<Word>
+
+    @Mock
+    private lateinit var testThrowable: Throwable
+
+    private lateinit var presenter: MainContract.Presenter
+
+    private lateinit var countWordsSubject: Subject<List<Word>>
+
+    private lateinit var startButtonClickedSubject: Subject<Any>
+
+    @Before
+    fun onBefore() {
+        countWordsSubject = BehaviorSubject.create()
+        startButtonClickedSubject = PublishSubject.create()
+
+        Mockito.`when`(countWordsUseCase.execute(anyString())).thenReturn(countWordsSubject)
+        Mockito.`when`(view.getStartButtonClickObservable()).thenReturn(startButtonClickedSubject)
+        Mockito.`when`(view.getUrlChangeObservable()).thenReturn(Observable.just(TEST_URL))
+
+        presenter = MainPresenter(countWordsUseCase)
+        presenter.bindView(view)
+    }
+
+    @After
+    fun onAfter() {
+        verify(view).getStartButtonClickObservable()
+        verify(view).getUrlChangeObservable()
+        verify(countWordsUseCase).execute(eq(TEST_URL))
+        verifyNoMoreInteractions(view, countWordsUseCase)
+    }
+
+    @Test
+    fun `given useCase does nothing on start button click should set Started viewState`() {
+        startButtonClickedSubject.onNext(Any())
+        verify(view).setViewState(eq(MainContract.ViewState.Started))
```

Comment:
Please keep one empty line between the `give`, `on`, `should` parts to improve readability. The notices is related to all tests.